parquet/src/arrow/array_reader/builder.rs

```diff
@@ -39,7 +39,47 @@ use crate::schema::types::{ColumnDescriptor, ColumnPath, Type};
 
 /// Builds [`ArrayReader`]s from parquet schema, projection mask, and RowGroups reader
 pub struct ArrayReaderBuilder<'a> {
+    /// Source of row group data
     row_groups: &'a dyn RowGroups,
+    /// Optional cache options for the array reader
+    cache_options: Option<&'a CacheOptions<'a>>,
+}
+
+/// Builder for [`CacheOptions`]
+#[derive(Debug, Clone)]
+pub struct CacheOptionsBuilder<'a> {
```

Comment:
My reasoning for making a builder is that it makes it clearer that the `consumer` and `producer` must be created in pairs